src/components/AllMovies/AllMovies.js

```diff
@@ -0,0 +1,47 @@
+import './AllMovies.css';
+import Card from '../Card/Card';
+import FilterRibbon from './FilterRibbon/FilterRibbon';
+import PropTypes from 'prop-types';
+import Loader from '../LoadSpinner/LoadSpinner';
+
+const AllMovies = ({  waitingForFetch, serverError, movies, changeSearch, searchValue  }) => {
+  let movieCards;
+
+  if(movies) {
+    movieCards = movies.map(movie => {
+      return (
+        <Card
+          movieTitle={movie.title}
+          moviePoster={movie.poster_path}
+          movieYear={movie.release_date.slice(0,4)}
+          key={movie.id}
+          id={movie.id}
+        />
+      )
+    })
+  }
+
+  const ErrorMessage = () => {
+    return (<h3 className='error-message' id='error-message'>Sorry, No movies to display</h3>);
+  }
+  
+  return (
+    <section className='all-movies-view'>
+      {waitingForFetch && <Loader />}
```

Comment:
Another pattern is to move this out of the return and return solo.

```suggestion
      if (waitingForFetch) return <Loader />;
```

Reply:
Can also combine if there are no movies.
if (waitingForFetch || !movies) return <Loader />;

Reply:
Evan, love that you pointed that out. So loader goes away if there's a response. Good or bad. Once there's a response, if no movies populate the data array, then another error message would be displayed. I did not want the loader to show up if there was an error in the request.

Reply:
Yeah you could also have another conditional after that checks for error and displays error screen. Over time you'll develop your style and organization.